src/test/java/io/tiledb/java/api/VFSTest.java

```diff
@@ -0,0 +1,322 @@
+
+  @Test
+  public void testVFSWithConfig() throws Exception {
+    try (Context ctx = new Context();
+        Config config = new Config()) {
+      // check default s3 scheme
+      Assert.assertEquals("https", config.get("vfs.s3.scheme"));
+      // set config to not use default s3 scheme
+      config.set("vfs.s3.scheme", "http");
+      Assert.assertEquals("http", config.get("vfs.s3.scheme"));
+      try (VFS vfs = new VFS(ctx, config);
+          Config vfsConfig = vfs.getConfig()) {
+        Assert.assertTrue(vfsConfig instanceof Config);
+        Assert.assertEquals("http", vfsConfig.get("vfs.s3.scheme"));
+      }
+    }
+  }
+
+  @Test
+  public void testVFSIsSupportedFs() throws Exception {
+    try (Context ctx = new Context();
+        VFS vfs = new VFS(ctx)) {
+      Assert.assertTrue(vfs.isSupportedFs("file://"));
+      Assert.assertTrue(vfs.isSupportedFs(new URI("file:///path/to/resource")));
+
+      Assert.assertEquals(vfs.isSupportedFs("s3://"), vfs.isSupportedFs(Filesystem.TILEDB_S3));
+
+      Assert.assertEquals(
+          vfs.isSupportedFs("s3"), vfs.isSupportedFs(new URI("s3://bucket/path/to/resource")));
+
+      Assert.assertEquals(vfs.isSupportedFs("hdfs://"), vfs.isSupportedFs(Filesystem.TILEDB_HDFS));
+
+      Assert.assertEquals(
+          vfs.isSupportedFs("hdfs"), vfs.isSupportedFs(new URI("hdfs:///path/to/resource")));
+    }
+  }
+
+  @Test
+  public void testVFSIsFile() throws Exception {
+    try (Context ctx = new Context();
+        VFS vfs = new VFS(ctx)) {
+      File fooFile = tmp.newFile("foo");
+      File barDir = tmp.newFolder("bar");
+
+      Assert.assertTrue(vfs.isFile(fooFile.toString()));
+      // workaround for TileDB #1097
+      Assert.assertTrue(vfs.isFile(URI.create("file://".concat(fooFile.getPath()))));
+
+      Assert.assertFalse(vfs.isFile(barDir.toString()));
+      // workaround for TileDB #1097
+      Assert.assertFalse(vfs.isFile(URI.create("file://".concat(barDir.getPath()))));
+    }
+  }
+
+  @Test
+  public void testVFSIsDir() throws Exception {
+    try (Context ctx = new Context();
+        VFS vfs = new VFS(ctx)) {
+      File fooFile = tmp.newFile("foo");
+      File barDir = tmp.newFolder("bar");
+
+      Assert.assertTrue(vfs.isDirectory(barDir.toString()));
+      // workaround for TileDB #1097
+      Assert.assertTrue(vfs.isDirectory(URI.create("file://".concat(barDir.getPath()))));
+
+      Assert.assertFalse(vfs.isDirectory(fooFile.toString()));
+      // workaround for TileDB #1097
+      Assert.assertFalse(vfs.isDirectory(URI.create("file://".concat(fooFile.getPath()))));
+    }
+  }
+
+  @Test
+  public void testVFSCreateDirectory() throws Exception {
+    try (Context ctx = new Context();
+        VFS vfs = new VFS(ctx)) {
+      String dirPath = Paths.get(tmp.getRoot().toString(), "foo").toString();
+      String newDir = vfs.createDirectory(dirPath);
+      Assert.assertTrue(vfs.isDirectory(newDir));
+    }
+  }
+
+  /* @Test(expected = TileDBError.class)
```

Comment:
Is this test suppose to be commented out?

Reply:
Yeah we don't check on creation if the resource already exists, I thought we did. Removed